src/components/auth/Login.test.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import Login from './Login';
+
+describe('login validation', () => {
+  test('successful login', async () => {
+    render(
+      <Router>
+        <Login />
+      </Router>,
+    );
+
+    userEvent.type(screen.getByLabelText('Username'), 'Demo');
+    userEvent.type(screen.getByLabelText('Password'), 'Password');
+
+    await waitFor(() => {
+      userEvent.click(screen.getByRole('button', { name: 'Login' }));
+      expect(screen.getByRole('button', { name: 'Login' })).toBeDisabled();
+    });
+  });
+
+  test('empty fields errors', async () => {
+    render(
+      <Router>
+        <Login />
+      </Router>,
+    );
+
+    await waitFor(() => {
```

Comment:
Instead of using waitFor callback, you can put the logic inside after the waitFor, since you're using await, it will hold the process until waitFor is done and then move to the next line, the same behavior you want here.